src/test/java/com/openshift/internal/restclient/ApiTypeMapperTest.java

```diff
@@ -32,10 +36,22 @@
 	private MocksFactory factory = new MocksFactory();
 	
 	@Test
+	public void testKubernetesResourceIsSupportedAfterInitiallyErrorIsThrown() throws Exception {
+		IService resource = factory.stub(ResourceKind.SERVICE);
+		try {
+			when(getHttpClient().get(eq(new URL(super.base + "/api")), anyInt())).thenThrow(new RuntimeException());
+			assertTrue("Exp. Kube support", mapper.isSupported(resource));
+		}catch(RuntimeException e) {
```

Comment:
@fbricon I didnt catch this before the merge. What do you mean it can't cover 'both' paths.  What is 'both' in this sentence.  The test is meant to catch the case where the internal collection is initialized before it is populated and then subsequent actions are unable to find the supported endpoints.